src/Ci/SourceHut.php

```diff
@@ -0,0 +1,60 @@
+<?php declare(strict_types=1);
+
+namespace OndraM\CiDetector\Ci;
+
+use OndraM\CiDetector\CiDetector;
+use OndraM\CiDetector\Env;
+use OndraM\CiDetector\TrinaryLogic;
+
+class SourceHut extends AbstractCi
+{
+    public static function isDetected(Env $env): bool
+    {
+        return ($env->get('JOB_ID') !== false) && $env->get('JOB_URL') !== false;
+    }
+
+    public function getCiName(): string
+    {
+        return CiDetector::CI_SOURCEHUT;
+    }
+
+    public function isPullRequest(): TrinaryLogic
+    {
+        return TrinaryLogic::createMaybe();
```

Comment:
Are all the environment variables the same if this is build for pull request? I'm just thinking whether there isn't some way how to detect whether this is pull request.

Reply:
SourceHut doesn't have merge requests the way GitHub and GitLab does. Instead, it uses mailing lists for patches. When you trigger a build by pushing a commit, you get an extra env var:

    BUILD_SUBMITTER=git.sr.ht

It's suppose to be possible to trigger a build by sending a patch to a mailing list, but I can't seem to get that to work right now. I imagine we could then check for this:

    BUILD_SUBMITTER=lists.sr.ht

But that then brings me back to my last comment -- this is tied to a particular instance, and that feels wrong. I really don't think there's any way to check for this unless another env var is added, like `CI_SOURCE=patch`

Let me ask @ddevault, but he may want to better understand use cases for detecting CI systems. It would help if you'd be able to tell me your motivation for writing ci-detector in the first place

I just want my build logs to look nicer, that's all 😂

Reply:
Ok, thanks. I appreciate your desire to improve the tools! :+1: 

Well I described some of the use-cases in [comment above](https://github.com/OndraM/ci-detector/pull/101#pullrequestreview-625773901).

Its not that uncommon especially for some CLI tools (usually QA tools) when they want to read properties of the current build - like build ID (for example for logging purposes), branch name,  git commit etc. Having many different CI servers in the wild, you want the tool to be portable among them. This means you need a way how to detect the CI server and its properties - which are provided by environment variables, unfortunately named differently on each of these CI servers. And this is the job CI detector, which contains adapter for each supported CI server.